sympy/discrete/convolutions.py

```diff
@@ -89,7 +90,30 @@ def convolution(a, b, cycle=0, dps=None, prime=None, dyadic=None, subset=None):
     elif subset:
         ls = convolution_subset(a, b)
     else:
-        ls = convolution_fft(a, b, dps=dps)
+        def loop(a):
+            dens = []
+            for i in a:
+                if isinstance(i, Rational) and not isinstance(i, Integer)r:
```

Comment:
```suggestion
                if isinstance(i, Rational) and not isinstance(i, Integer):
```

Reply:
Good spot, thanks, though do you think `i.is_integer` would be better here?